packages/ember-htmlbars/tests/integration/component_invocation_test.js

```diff
@@ -575,6 +575,44 @@ QUnit.test("comopnent should rerender when a property is changed during children
 
 });
 
+QUnit.test("components in template of a yielding component should have the proper parentView", function() {
+  var outer, innerTemplate, innerLayout;
+
+  registry.register('component:x-outer', Component.extend({
+    init() {
+      this._super(...arguments);
```

Comment:
Fun fact: This transpiles directly to:

```
this._super.apply(this, arguments);
```

Which is **awesome**. Elsewhere in ember we use (and I am far less crazy about):

```
registry.register('component:x-outer', Component.extend({
   init(...args) {
    this._super(...args);
    outer = this;
      }
 }));
```

Which builds an intermediate array of args before unrolling.